x/auth/ante.go

```diff
@@ -20,10 +19,13 @@ func NewAnteHandler(accountMapper sdk.AccountMapper) sdk.AnteHandler {
 					sdk.ErrUnrecognizedAddress(payerAddr).Result(),
 					true
 			}
-			// TODO: Charge fee from payerAcc.
-			// TODO: accountMapper.SetAccount(ctx, payerAddr)
+			coins := payerAcc.GetCoins().Minus(tx.GetFees())
```

Comment:
You don't check if they have enough coins to pay the fee

Reply:
A few test cases would be a good start. Like a table test to hit all the conditions.